pkg/aws/subnets.go

```diff
@@ -0,0 +1,67 @@
+package aws
+
+import "github.com/aws/aws-sdk-go/service/ec2"
+
+var (
+	tagSubmarinerGatgeway = ec2Tag("submariner.io/gateway", "")
+	tagInternalELB        = ec2Tag("kubernetes.io/role/internal-elb", "")
+)
+
+func (ac *awsCloud) findPublicSubnet(vpcID string, filter *ec2.Filter) (*ec2.Subnet, error) {
+	filters := []*ec2.Filter{
+		ec2Filter("vpc-id", vpcID),
+		ac.filterByCurrentCluster(),
+		filter,
+	}
+
+	result, err := ac.client.DescribeSubnets(&ec2.DescribeSubnetsInput{Filters: filters})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.Subnets) == 0 {
+		return nil, newNotFoundError("public subnet")
+	}
+
+	return result.Subnets[0], nil
```

Comment:
I think we need to check whether we can create a machine with the specified machine type in this subnet, we have encountered an issue that the returned subnet does not support the machine type that we specified.

you may use `DescribeInstanceTypeOfferings` api to check, there is an example: https://github.com/open-cluster-management/submariner-addon/blob/main/pkg/cloud/aws/aws.go#L348

Reply:
I'm looking into it already, thanks for noticing :)

Reply:
I've addressed this in #5